Settings: Introduce a ownership information section to SiteOwnership

tyxla opened the pull request:
This PR introduces a section to explain what **Site owner** and **Plan purchaser** are. It follows the designs, suggested in p6TEKc-29d-p2, and is intended to land after #25725, which introduces the UI to allow Jetpack plan ownership to be transferred to another user.

Preview:

![](https://cldup.com/EoGA3SsHeq.png)

To test:
* Checkout this branch.
* Go to `http://calypso.localhost:3000/settings/manage-connection/:site`, where `:site` is a Jetpack site.
* Verify the new section is at the bottom of the **Site ownership** card, and it looks as shown on the screenshot.
* Try enabling the Happychat staging HUD to simulate that chats are available, and refresh the page; verify the chat link appears properly.
* Verify it works well on mobile devices.

tyxla commented:
@joanrho I've used the copy from the mockup, but I'm not sure if that's the latest approved version of it. Could we confirm this?

sirreal commented on client/components/forms/form-setting-explanation/style.scss:
```diff
@@ -9,4 +9,17 @@
+		font-weight: inherit;
+		line-height: inherit;
+		padding: 0;
+	}
```
Are all these style OK in the shared component or should they be in the specific styles?

tyxla replied:
These are pretty specific to buttons that are children of `FormSettingExplanation`, so I chose to keep them here.

sirreal commented on client/my-sites/site-settings/manage-connection/ownership-information.jsx:
```diff
@@ -0,0 +1,60 @@
+							'their clients may be Plan purchasers who use their billing information to purchase the plan ' +
+							'for the site.'
+					) }
+					<HappychatButton>{ translate( 'Need help? Chat with us' ) }</HappychatButton>
```
This likely needs a line break and or whitespace. I've shared a capture.

tyxla replied:
👍 I've addressed that in fa08543fc3b0fc8aa7db18061f2aa3cca47f80cf

See the updated preview:

![](https://cldup.com/fqNIidYtZI.png)

tyxla commented:
Thanks for the review @sirreal! Addressed your feedback, this one is ready for another review.

> I'm unable to verify the correct Happychat queue or that Happy chat was working correctly. I'm not sure if that's related to Happychat staging or another problem. I did not receive chats initiated as I expected. That would be worth verifying on staging.

👍
